Add default label to Fleet Secrets to enable BRO backups

mallardduck opened the pull request:
<!-- This template is for Devs to give QA details before moving the issue To-Test -->
### Summary
This PR fixes SURE-6919
Fixes https://github.com/rancher/dashboard/issues/11211
<!-- Define findings related to the feature or bug issue. -->

There were recent fixes to the BRO codebase that corrected a bug which backed up far more data than it should have been backing up. In other terms, the resourceset filters BRO uses to collect resources for backups are properly followed now.

As such, the only fleet secrets that get backed up would be ones with the `"fleet.cattle.io/managed=true"` label. Since that is the existing rule defined for fleet within the resoruceset.

### Technical notes summary
<!-- Outline technical changes which may pass unobserved or may help to understand the process of solving the issue -->
This PR aims to make sure that the secrets being created by the fleet UI are labeled as managed by fleet. This in turn will allow BROs rules to collect the secrets for backup.

This change is not retroactive - any existing secrets should be updated by Rancher admins to include the correct label.

Additionally, we are intentionally not addressing when users select a secret created outside of the UI for fleet. This maybe worth looking into expanding on later, but for now the BRO team perspective is that those would be "user-created" (and managed) resources; not something Rancher created for them.

### Areas or cases that should be tested
<!-- Areas that should be tested can include Airgap checks, Rancher upgrades, K8s upgrade, etc. -->
<!-- Which browser did you use for local testing? The reviewer should test with a different browser. -->
<!-- Add missing steps or rewrite them if have been missed or to complement existing information. This should define a clear way to reproduce it and not an approximation. -->
1. Create a GitRepo and define a new secret via the GitRepo UI,
2. Find the secrets name after creation,
3. Explore secrets in the local cluster to find new secret,
4. Observe it currently lacks the expected label.

After this PR it should be observed that the secret will have the expected label.
Once complete this will allow testing of the main issue per direction on https://github.com/rancher/dashboard/issues/11211

### Areas which could experience regressions
<!-- Create a detailed list of areas to be analyzed which may be affected by the changes, which would require a prior research to avoid regressions. -->
Unlikely to occur.

### Screenshot/Video
<!-- Attach screenshot or video of the changes and eventual comparison if you find it necessary -->
N/A. There are no visual UX changes.

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone <!-- The milestone should automatically be assigned if the linked issue has one, but does not always happen (incorrectly linked, issue has no milestone, etc) -->
- [x] The PR template has been filled out
- [x] The PR has been self reviewed <!-- There are no TODOs, no incorrect files in the PR, all the required files are there, no commented out code, etc-->
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or **there are no UX changes**

richard-cox commented on shell/edit/fleet.cattle.io.gitrepo.vue:
```diff
@@ -413,8 +413,9 @@ export default {
-            generateName: 'auth-'
-          },
+            generateName: 'auth-',
+            labels:       { 'fleet.cattle.io/managed': 'true' }
```
This should reference https://github.com/rancher/dashboard/blob/53544f5d190ba53dd7b6c819208732efe30c2ee1/shell/config/labels-annotations.js#L113

mallardduck replied:
Appreciate it - fixed this now.

richard-cox reviewed (changes requested):
@mallardduck Thanks for creating this PR. Can you make sure you run through the PR checklist?

nwmac commented:
This also needs automated tests to checks that the annotation is applied in the API request that the UI sends.

mallardduck commented on shell/edit/fleet.cattle.io.gitrepo.vue:
```diff
@@ -22,7 +22,7 @@ import { base64Decode, base64Encode } from '@shell/utils/crypto';
 import { _CREATE } from '@shell/config/query-params';
 import { isHarvesterCluster } from '@shell/utils/cluster';
-import { CAPI, CATALOG } from '@shell/config/labels-annotations';
+import { CAPI, CATALOG, FLEET as FLEET_LABELS } from '@shell/config/labels-annotations';
```
Used `as` since there's already a `FLEET` import.

mallardduck commented:
So pending successful run of `e2e-test (admin, @adminUser, @fleet)` which will have the tests I modified to verify the new label - I think this is good for review now. I adjusted the test to verify that the request being sent and the secret in the response both have the label we expect. Assuming I read the chai docs right then I expect the new assertion on the tests will be valid now, so I'll check that box as well.

mallardduck commented:
Added @joshmeranda as a Reviewer for O&B team perspective.

joshmeranda reviewed (approved):
LGTM, obviously would like to get eyes from UI

richard-cox reviewed (approved):
I've created https://github.com/rancher/dashboard/issues/11222 for some follow on UX changes. Currently we hide all system based labels and annotations, and in some cases it would be good to see them
